apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx

```diff
@@ -105,8 +110,38 @@ class VisualizerModal extends React.Component {
     return columns.filter(column => isColumnNumeric(records, column));
   });
 
+  getValuesForFilterColumn = memoize((records, column) => {
+    if (!records || !column) {
+      return [];
+    }
+    let values = Array.from(new Set(records.map(record => record[column])));
```

Comment:
Clever! I did a little digging to see if this was common and it looks like it is.  The only difference I found is that it might be more common to express this operation with the spread operator:

```js
let values = [...new Set(records.map(record => record[column]))];
```

I think yours reads cleaner though.